docs/tutorials/storage.rst

```diff
@@ -0,0 +1,231 @@
+
+    with open('story.txt') as my_file:
+        content = my_file.read()
+    print(content)
+
+The ``with`` statement uses the ``open`` function to open a file and assign it
+to an object. In the example above, the ``open`` function opens the file called
+``story.txt`` (obviously a text file containing a story of some sort).
+The object that's used to represent the file in the Python code is called
+``my_file``. Subsequently, in the code block indented underneath the ``with``
+statement, the ``my_file`` object is used to ``read()`` the content of the
+file and assign it to the ``content`` object.
+
+Here's the important point, *the next line containing the* ``print`` *statement
+is not indented*. The code block associated with the ``with`` statement is only
+the single line that reads the file. Once the code block associated with the
+``with`` statement is closed then Python (and MicroPython) will automatically
+close the file for you. This is called context handling and the ``open``
+function creates objects that are context handlers for files.
+
+Put simply, the scope of your interaction with a file is defined by the code
+block associated with the ``with`` statement that opens the file.
+
+Confused?
+
+Don't be. I'm simply saying your code should look like this::
+
+    with open('some_file') as some_object:
+        # Do stuff with some_object in this block of code
+        # associated with the with statement.
+
+    # When the block is finished then MicroPython
+    # automatically closes the file for you.
+
+Just like a paper file, a digital file is opened for two reasons: to read its
+content (as demonstrated above) or to write something to the file. The default
+mode is to read the file. If you want to write to a file you need to tell the
+``open`` function in the following way::
+
+    with open('hello.txt', 'w') as my_file:
+        my_file.write("Hello, World!")
+
+Notice the ``'w'`` argument is used to set the ``my_file`` object into write
+mode. You could also pass an ``'r'`` argument to set the file object to read
+mode, but since this is the default, it's often left off.
+
+Writing data to the file is done with the (you guessed it) ``write``
+method that takes the string you want to write to the file as an argument. In
+the example above, I write the text "Hello, World!" to a file called
+"hello.txt".
+
+Simple!
+
+.. note::
+
+    When you open a file and write (perhaps several times while the file is
+    in an open state) you will be writing OVER the content of the file if it
+    already exists.
+
+    If you want to append data to a file you should first read it, store the
+    content somewhere, close it, append your data to the content and then open
+    it to write again with the revised content.
+
+    While this is the case in MicroPython, "normal" Python can open
+    files to write in "append" mode. That we can't do this on the micro:bit is
+    a result of the simple implementation of the file system.
```

Comment:
Ok.  I thought it might be as simple as re-opening the file (and refilling caches, etc) like it was never closed from the last time it was written to.